### PR TITLE
samples: add support for updateDatabase in Cloud Spanner

### PR DESCRIPTION
**Pull request by aayushimalik**

### CODE REVIEW

**Review by rajatbhatta — COMMENTED** [review]
@aayushimalik: Could you please use the sample shared internally in the client spec doc to write this sample? It is to maintain consistency in the sample across clients.

File: samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      final DatabaseAdminClient databaseAdminClient = spanner.getDatabaseAdminClient();
+
+      // Enable drop protection
+      Database update_to = databaseAdminClient.newDatabaseBuilder(
+              DatabaseId.of(projectId, instanceId, databaseId))
+          .enableDropProtection().build();
+      OperationFuture<Database, UpdateDatabaseMetadata> operation = databaseAdminClient.updateDatabase(
+          update_to, DatabaseField.DROP_PROTECTION);
+      System.out.printf("Waiting for update operation for %s to complete...\n", databaseId);
+      Database db = operation.get();
```
Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:52]: Please add a timeout value to the .get() call based on the time it usually takes for the LRO to complete.

File: samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      System.out.printf("Updated database %s.\n", databaseId);
```
Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:53]: We can try to get databaseId from the `db` object retrieved as `db.getId().getName()`.

Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:53]: Can we add the full DatabaseName here in the print statement? There are two ways you could do this:

1. Declare `dbId = DatabaseId.of(projectId, instanceId, databaseId)` and use it while creating the `databaseToUpdate` object and `dbId.getName()` should give you the databaseName to print here.
2. From output of operation.get() `db`: `db.getId().getName()`.

File: samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
```
Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:56]:
```suggestion
      throw SpannerExceptionFactory.asSpannerException(e.getCause());
```

File: samples/snippets/src/test/java/com/example/spanner/UpdateDatabaseSampleIT.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        .get(5, TimeUnit.MINUTES);
+
+    // Runs sample
+    final String out = SampleRunner.runSample(() -> UpdateDatabaseSample
+        .updateDatabase(projectId, instanceId, databaseId)
+    );
+
+    assertTrue(
+        "Expected that database would have been updated. Output received was " + out,
+        out.contains(String.format("Updated database %s", databaseId))
```
Comment (rajatbhatta) [inline samples/snippets/src/test/java/com/example/spanner/UpdateDatabaseSampleIT.java:45]: This should also be changed to databaseName.

File: samples/snippets/src/test/java/com/example/spanner/UpdateDatabaseSampleIT.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    );
+
+    // Cleanup
+    Database update_to = databaseAdminClient.newDatabaseBuilder(
```
Comment (rajatbhatta) [inline samples/snippets/src/test/java/com/example/spanner/UpdateDatabaseSampleIT.java:49]:
```suggestion
    Database databaseToUpdate = databaseAdminClient.newDatabaseBuilder(
```

File: samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java (hunk 1 above)
Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:52]:
```suggestion
      Database updatedDb = operation.get();
```

Comment (rajatbhatta) [reply samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:52]: operation.get(5, TimeUnit.MINUTES);

File: samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+      // Enable drop protection
+      DatabaseId dbId = DatabaseId.of(projectId, instanceId, databaseId);
+      Database databaseToUpdate =
+          databaseAdminClient.newDatabaseBuilder(dbId).enableDropProtection().build();
+      OperationFuture<Database, UpdateDatabaseMetadata> operation =
+          databaseAdminClient.updateDatabase(databaseToUpdate, DatabaseField.DROP_PROTECTION);
+      System.out.printf("Waiting for update operation for %s to complete...\n", dbId);
+      Database updatedDb = operation.get(5, TimeUnit.MINUTES);
+      System.out.printf("Updated database %s.\n", updatedDb.getId().getName());
```
Comment (rajatbhatta) [inline samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:51]:
```suggestion
      System.out.println("Updated database %s.", updatedDb.getId().getName());
```

Comment (aayushimalik) [reply samples/snippets/src/main/java/com/example/spanner/UpdateDatabaseSample.java:50]: We can't use the formatting with println, we'd have to use String.format then. I feel using "\n" is more convenient.

**Pull request merged**